src/InnerApi/Identity/Authorization/BearerTokenMiddleware.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using Microsoft.IdentityModel.Tokens;
+
+namespace SFA.DAS.Apprenticeships.InnerApi.Identity.Authorization
+{
+    /// <summary>
+    /// Middleware that handles the claim values from bearer tokens in incoming requests
+    /// </summary>
+    public class BearerTokenMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly IConfiguration _configuration;
+
+        public BearerTokenMiddleware(RequestDelegate next, IConfiguration configuration)
+        {
+            _next = next;
+            _configuration = configuration;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            if (DisableAccountAuthorisation())
+            {
+                await _next(context);
+                return;
+            }
+
+            RequireClaimsValidation(context);
+            var token = ReadTokenFromAuthHeader(context);
+            var principal = ValidateToken(token);
+
+            if (!HandleProviderAccountClaim(context, principal) && !HandleEmployerAccountClaim(context, principal))
+            {
+                throw new UnauthorizedAccessException();
+            }
+
+            await _next(context);
+        }
+
+        private bool DisableAccountAuthorisation()
+        {
+            return bool.TryParse(_configuration["DisableAccountAuthorisation"], out var disableAccountAuthorisation) && disableAccountAuthorisation;
```

Comment:
we seem to be using a combination of american and british english for authorisation/authorization, worth standardising? I think elsewhere we are using z to match the c# standard